docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,259 @@
+
+# List only the models from the spaCy library.
+>>> api.list_models(filter="spacy")
+```
+
+Explore available public datasets with `list_datasets`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List only text classification datasets.
+>>> api.list_datasets(filter="task_categories:text-classification")
+
+# List only datasets in Russian for language modeling.
+>>> api.list_datasets(filter=("languages:ru", "task_ids:language-modeling"))
+```
+
+### Inspect model or dataset metadata
+
+Get important information about a model or dataset as shown in the following:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# Get metadata of a single model.
+>>> api.model_info("distilbert-base-uncased")
+
+# Get metadata of a single dataset.
+>>> api.dataset_info("glue")
+```
+
+### Delete a repository
+
+Delete a repository with `delete_repo`. Make sure you are certain you want to delete a repository because this is an irreversible process!
+
+1. Get your Hugging Face API token:
+
+```python
+>>> from huggingface_hub import HfFolder
+>>> folder = HfFolder()
+>>> token = folder.get_token()
+```
+
+2. Give `delete_repo` your token and the name of the repository to delete:
+
+```python
+>>> api.delete_repo(token=YOUR_HF_API_TOKEN name=REPO_NAME)
+```
+
+Delete a dataset repository by adding the `type` parameter:
+
+```python
+>>> api.create_repo(token=YOUR_HF_API_TOKEN, name=REPO_NAME, type="dataset")
+```
+
+### Change repository visibility
+
+A repository can be public or private. A private repository is only visible to you or members of your organization. Change a repository to private as shown in the following:
+
+```python
+>>> api.update_repo_visibility(token=YOUR_HF_API_TOKEN, name=REPO_NAME, private=True)
+```
+
+## `Repository` 
+
+The `Repository` class allows you to push models or other repositories to the Hub. `Repository` is a wrapper over Git and Git-LFS methods, so make sure you have Git-LFS installed (see [here](https://git-lfs.github.com/) for more instructions) and set up before you begin. If you are already familiar with common Git commands, then the `Repository` class should feel familiar. 
+
+Start by instantiating a `Repository` object with a path to a local Git clone or repository:
+
+```python
+>>> from huggingface_hub import Repository
+>>> repo = Repository(local_dir="<path>/<to>/<folder>")
+```
+
+### Clone a repository
+
+The `clone_from` parameter clones a repository from a specified directory using a URL (if you are working offline, this parameter should be `None`):
+
+```python
+>>> repo = Repository(local_dir="huggingface-hub", clone_from="https://github.com/huggingface/huggingface_hub")
+```
+
+This parameter can also clone a repository from a Hugging Face model ID:
+
+```python
+>>> repo = Repository(local_dir="w2v2", clone_from="facebook/wav2vec2-large-960h-lv60")
+```
+
+If you want to commit or push to a cloned repository that belongs to you or your organizations:
```

Comment:
This could likely be its own section since it's not abut cloning repositories. Maybe let's just add a header here